### ISSUE TITLE
RMSProp doesn't work for MobileNetV2 on ImageNet

### ISSUE THREAD

**Issue by hetong007** [apache/incubator-mxnet#10320]
## Description
Trying to train MobileNetV2_1.0 on ImageNet-1k with RMSProp, but the model is not effectively trained, i.e. the top-1 accuracy stays at around 0.1%.

## Environment info (Required)

```
----------Python Info----------
('Version      :', '2.7.12')
('Compiler     :', 'GCC 5.4.0 20160609')
('Build        :', ('default', 'Nov 20 2017 18:23:56'))
('Arch         :', ('64bit', 'ELF'))
------------Pip Info-----------
('Version      :', '9.0.1')
('Directory    :', '/usr/local/lib/python2.7/dist-packages/pip')
----------MXNet Info-----------
('Version      :', '1.2.0')
('Directory    :', '/usr/local/lib/python2.7/dist-packages/mxnet')
('Commit Hash   :', '0ff490b62e33b1d33f5527de3a22d5f54b60fffc')
----------System Info----------
('Platform     :', 'Linux-4.4.0-1052-aws-x86_64-with-Ubuntu-16.04-xenial')
('system       :', 'Linux')
('node         :', 'ip-172-31-3-42')
('release      :', '4.4.0-1052-aws')
('version      :', '#61-Ubuntu SMP Mon Feb 12 23:05:58 UTC 2018')
----------Hardware Info----------
('machine      :', 'x86_64')
('processor    :', 'x86_64')
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                32
On-line CPU(s) list:   0-31
Thread(s) per core:    2
Core(s) per socket:    16
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 79
Model name:            Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz
Stepping:              1
CPU MHz:               2699.894
CPU max MHz:           3000.0000
CPU min MHz:           1200.0000
BogoMIPS:              4600.10
Hypervisor vendor:     Xen
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              46080K
NUMA node0 CPU(s):     0-31
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single retpoline kaiser fsgsbase bmi1 hle avx2 smep bmi2 erms invpcid rtm rdseed adx xsaveopt
----------Network Test----------
Setting timeout: 10
Timing for MXNet: https://github.com/apache/incubator-mxnet, DNS: 0.0023 sec, LOAD: 0.5335 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.0224 sec, LOAD: 0.5653 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 0.0283 sec, LOAD: 0.1515 sec.
Timing for Conda: https://repo.continuum.io/pkgs/free/, DNS: 0.0098 sec, LOAD: 0.3304 sec.
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 0.1106 sec, LOAD: 0.0703 sec.
Timing for Gluon Tutorial(cn): https://zh.gluon.ai, DNS: 0.0884 sec, LOAD: 0.1099 sec.
```

Package used (Python/R/Scala/Julia):
python, installed with `pip install mxnet-cu90 --pre` on Mar 29th, 2018.

## Steps to reproduce

On p3.8xlarge:

1. Get the script from https://github.com/zhreshold/mxnet/blob/2fbfdbcbacff8b738bd9f44e9c8cefc84d6dfbb5/example/gluon/train_imagenet.py
2. Replace line 133-136 with
```
    trainer = gluon.Trainer(net.collect_params(), 'rmsprop',
                            {'learning_rate': args.lr, 'wd': args.wd,
                            'centered': True},
                            kvstore = args.kvstore)
```
3. Run with `python3 train_imagenet.py --data ~/data/ILSVRC2012/ --model mobilenetv2_1.0 -j 32 --gpus 0,1,2,3 -b 384 --lr 0.0045 --weight-decay 0.00004`

Here's the results from the first few batches. 

```
INFO:root:Namespace(batch_size=384, data='/home/ubuntu/data/ILSVRC2012/', dtype='float32', epochs=120, gpus='0,1,2,3', kvstore='device', log_interval=10, lr=0.0045, lr_factor=0.1, lr_steps='30,60,90', model='mobilenetv2_1.0', momentum=0.9, num_workers=32, prefix='', pretrained=False, resume='', save_frequency=10, start_epoch=0, wd=4e-05)
INFO:root:Loading image folder /home/ubuntu/data/ILSVRC2012/train, this may take a bit long...
INFO:root:Loading image folder /home/ubuntu/data/ILSVRC2012/val, this may take a bit long...
[19:16:49] src/operator/nn/./cudnn/./cudnn_algoreg-inl.h:107: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
INFO:root:Epoch[0] Batch[9]	Speed: 864.038059 samples/sec	accuracy=0.002604,top_k_accuracy_5=0.006510
INFO:root:Epoch[0] Batch[19]	Speed: 666.248070 samples/sec	accuracy=0.001302,top_k_accuracy_5=0.006380
INFO:root:Epoch[0] Batch[29]	Speed: 624.697316 samples/sec	accuracy=0.001128,top_k_accuracy_5=0.005035
INFO:root:Epoch[0] Batch[39]	Speed: 435.179242 samples/sec	accuracy=0.001107,top_k_accuracy_5=0.005143
INFO:root:Epoch[0] Batch[49]	Speed: 339.314886 samples/sec	accuracy=0.001094,top_k_accuracy_5=0.005313
INFO:root:Epoch[0] Batch[59]	Speed: 439.406339 samples/sec	accuracy=0.001302,top_k_accuracy_5=0.005208
INFO:root:Epoch[0] Batch[69]	Speed: 345.037007 samples/sec	accuracy=0.001265,top_k_accuracy_5=0.005097
INFO:root:Epoch[0] Batch[79]	Speed: 246.770412 samples/sec	accuracy=0.001237,top_k_accuracy_5=0.004948
INFO:root:Epoch[0] Batch[89]	Speed: 404.376028 samples/sec	accuracy=0.001186,top_k_accuracy_5=0.004948
INFO:root:Epoch[0] Batch[99]	Speed: 526.128250 samples/sec	accuracy=0.001250,top_k_accuracy_5=0.004870
INFO:root:Epoch[0] Batch[109]	Speed: 372.590617 samples/sec	accuracy=0.001160,top_k_accuracy_5=0.004759
INFO:root:Epoch[0] Batch[119]	Speed: 522.334855 samples/sec	accuracy=0.001194,top_k_accuracy_5=0.004926
```

After the first epoch, the result is:
```
INFO:root:[Epoch 0] Training: accuracy=0.002422,top_k_accuracy_5=0.011254
INFO:root:[Epoch 0] Training time cost: 3235.584762
INFO:root:[Epoch 0] Validation: accuracy=0.001000,top_k_accuracy_5=0.005000
INFO:root:[Epoch 0] Saving checkpoint to mobilenetv2_1.0_best.params with Accuracy: 0.0010
```

## What have you tried to solve it?

1. Tried both `centered=True` and `centered=False`, there's no difference.
2. Tried to use `sgd`, the script has effect as the accuracy raises significantly.

**Comment by sxjscience** [comment]
@szhengac

**Comment by szhengac** [comment]
It is possible that the iterate stuck in some valley. You may try either a significantly larger stepsize or a much smaller one @hetong007 Also, can you try Adam or FTML?

**Comment by hetong007** [comment]
Seems that RMSProp just does not always work, as also reported here: https://github.com/tonylins/pytorch-mobilenet-v2

**Comment by hetong007** [comment]
@jonmorton thanks, will test! It's just not mentioned in Google's paper.
